Xamarin.Essentials/WebAuthenticator/WebAuthenticator.ios.tvos..cs

```diff
@@ -0,0 +1,179 @@
+                }
+
+                if (UIDevice.CurrentDevice.CheckSystemVersion(9, 0))
+                {
+                    var controller = new SFSafariViewController(new NSUrl(url.OriginalString), false)
+                    {
+                        Delegate = new NativeSFSafariViewControllerDelegate
+                        {
+                            DidFinishHandler = (svc) =>
+                            {
+                                // Cancel our task if it wasn't already marked as completed
+                                if (!(tcsResponse?.Task?.IsCompleted ?? true))
+                                    tcsResponse.TrySetException(new OperationCanceledException());
+                            }
+                        },
+                    };
+
+                    currentViewController = controller;
+                    await Platform.GetCurrentUIViewController().PresentViewControllerAsync(controller, true);
+                    return await tcsResponse.Task;
+                }
+#endif
+
+                var opened = UIApplication.SharedApplication.OpenUrl(url);
+                if (!opened)
+                    tcsResponse.TrySetException(new Exception("Error opening Safari"));
+            }
+            catch (Exception ex)
+            {
+                tcsResponse.TrySetException(ex);
+            }
+
+            return await tcsResponse.Task;
+        }
+
+        internal static bool OpenUrl(Uri uri)
+        {
+            // If we aren't waiting on a task, don't handle the url
+            if (tcsResponse?.Task?.IsCompleted ?? true)
+                return false;
+
+            try
+            {
+                // If we can't handle the url, don't
+                if (!WebUtils.CanHandleCallback(redirectUri, uri))
+                    return false;
+
+                currentViewController?.DismissViewControllerAsync(true);
+                currentViewController = null;
+
+                tcsResponse.TrySetResult(new AuthResult(uri));
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+            return false;
+        }
+
+        public static bool VerifyHasUrlSchemeOrDoesntRequire(string scheme)
+        {
+            // iOS11+ uses sfAuthenticationSession which handles its own url routing
+            if (UIDevice.CurrentDevice.CheckSystemVersion(11, 0))
+                return true;
+
+            var cleansed = scheme.Replace("://", string.Empty);
+            var schemes = GetCFBundleURLSchemes().ToList();
+            return schemes.Any(x => x.Equals(cleansed, StringComparison.InvariantCultureIgnoreCase));
+        }
+
+        static IEnumerable<string> GetCFBundleURLSchemes()
+        {
+            NSObject nsobj = null;
+            if (!NSBundle.MainBundle.InfoDictionary.TryGetValue((NSString)"CFBundleURLTypes", out nsobj))
+                yield return null;
```

Comment:
Are you actually wanting to return null, or are you looking for `yield break;`?

Reply:
Reworking this to make it more readable.